src/Renci.SshNet/BaseClient.cs

```diff
@@ -192,7 +198,6 @@ private protected BaseClient(ConnectionInfo connectionInfo, bool ownsConnectionI
             _connectionInfo = connectionInfo;
             _ownsConnectionInfo = ownsConnectionInfo;
             _serviceFactory = serviceFactory;
-            _logger = SshNetLoggingConfiguration.LoggerFactory.CreateLogger(GetType());
```

Comment:
The new Logger property is going to instantiate a new logger object on each access. Could it instead keep this line, instantiating the field from `connectionInfo.LoggerFactory`, with a `??` fallback to the static factory like you have done in Session?